From e3e88af8847a4269fb275fd054fc36c5066d5d49 Mon Sep 17 00:00:00 2001
From: Assaf Gordon <assafgordon@gmail.com>
Date: Mon, 15 Sep 2014 19:08:20 +0000
Subject: [PATCH 08/16] virt-what.in: warn about missing cpuid-virt helper
 program

Try to find the binary program on the $PATH, and give informative error
message if it wasn't found.
---
 virt-what.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/virt-what.in b/virt-what.in
index 3feeb60..5ae1ccb 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -80,6 +80,12 @@ if [ "x$root" = "x" ] && [ "$EFFUID" -ne 0 ]; then
     fail "this script must be run as root"
 fi
 
+# Try to locate the CPU-ID helper program
+CPUID_HELPER=$(which virt-what-cpuid-helper 2>/dev/null)
+if [ -z "$CPUID_HELPER" ] ; then
+    fail "virt-what-cpuid-helper program not found in \$PATH"
+fi
+
 # Many fullvirt hypervisors give an indication through CPUID.  Use the
 # helper program to get this information.
 
-- 
2.3.1

